src/super_context.ml

```diff
@@ -582,9 +584,12 @@ module Action = struct
           end
         | Some ("lib-available", lib) ->
           add_lib_dep acc lib Optional;
-          Some (str_exp (string_of_bool (Libs.lib_is_available sctx ~from:dir lib)))
+          Some (str_exp (string_of_bool (
+            (* XXX should we really be using the required_by of scope here? lib
+               isn't really required here, but optional *)
```

Comment:
I guess it doesn't matter since `lib_is_availale` won't raise. If it did we should set `required_by` to point to `loc`.

Reply:
It won't raise but it will populate the findlib cache with the execption I think. So if the library is actually required elsewhere, it will see the already populated exception from this call. At least this is how I think it will work, I haven't tested it.

Reply:
Hmm, that's true. I guess the best would be to update the required_by field when raising the exception, but I'm not sure how invasive this change is as the `required_by` in the error is a mix of the local stack and the external one. We can leave that as it for now